staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -92,6 +92,8 @@ func IsProbableEOF(err error) bool {
 		return true
 	case msg == "http: can't write HTTP request on broken connection":
 		return true
+	case msg == "http2: client connection lost":
```

Comment:
This message is returned when an http2 client hits a ping error and closes the connection. Returning true for IsProbableEOF in that scenario lowers the log level of the error to trace level, and makes watches and get requests retry

Reply:
is there a way to add a test that would notice if the message returned by `closeForLostPing()` changes and drifts so this no longer matches?

Reply:
Do you mean that as long as it is an error returned by closeForLostPing(), it should return True?

Reply:
right

Reply:
There's no test that ensures when a connection is closed because a ping is lost, that the returned error is identified as probably EOF. If the golang stdlib message changes, this will break silently. That's what I would like to see a test to catch.